### PR TITLE
guest: inlines host.GetConfig to config.Get

### PR DESCRIPTION
**Pull request by codefromthecrypt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This inlines `host.GetConfig()` to `config.Get()` after noticing logging will be in another package. This is simpler than describing a new term "host" which is only used to get configuration.

#### Which issue(s) this PR fixes:

NONE

#### Special notes for your reviewer:

This also does refactoring to sketch out how to test logic that requires wasm imports.

#### Does this PR introduce a user-facing change?

NONE

#### What are the benchmark results of this change?

```benchstat
goos: darwin
goarch: arm64
pkg: sigs.k8s.io/kube-scheduler-wasm-extension/internal/e2e/scheduler
                          │ before.txt  │              now.txt              │
                          │   sec/op    │   sec/op     vs base              │
Example_NodeNumber/New-12   73.68m ± 2%   75.34m ± 2%       ~ (p=0.093 n=6)
Example_NodeNumber/Run-12   46.12µ ± 4%   46.82µ ± 1%       ~ (p=0.394 n=6)
geomean                     1.843m        1.878m       +1.88%
```

### CODE REVIEW

File: guest/internal/mem/mem.go [hunk 1]
```diff
@@ -68,27 +68,6 @@ func Update(
 	return updater(readBuf)
 }
 
-func GetBytes(fn func(ptr uint32, limit BufLimit) (len uint32)) []byte {
```
Comment (codefromthecrypt) [inline guest/internal/mem/mem.go:71]: I removed this because it is difficult to test.

File: guest/config/config.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+package config
+
+import (
+	"unsafe"
+
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/mem"
+)
+
+var (
+	readConfig func() string = func() string {
```
Comment (codefromthecrypt) [inline guest/config/config.go:27]: ok I reworked this to add unit tests. This is the first time it is possible to test the logic around host imports.

**Comment by codefromthecrypt** [comment]
oops one more thing to settle

**Review by sanposhiho — APPROVED** [review]
👍 

/lgtm
/approve

**Pull request merged**